force_gromacs/io/gromacs_topology_writer.py

```diff
@@ -1,38 +1,41 @@
 import os
 
-from traits.api import List, Unicode, Dict, Int
+from traits.api import (
+    HasTraits, List, Str, Dict, Int, provides
+)
 
-from force_gromacs.core.base_gromacs_process import BaseGromacsProcess
+from force_gromacs.core.i_process import IProcess
 
 
-class GromacsTopologyWriter(BaseGromacsProcess):
+@provides(IProcess)
+class GromacsTopologyWriter(HasTraits):
     """  Class writes Gromacs topology file"""
 
     # --------------------
     #  Required Attributes
     # --------------------
 
     #: Reference name for the Gromacs simulation
-    sim_name = Unicode()
+    sim_name = Str()
```

Comment:
(Question: is this a simulation name, or rather a simulation ID?)

Reply:
At the moment it acts as a human-readable reference for the simulation data, rather than a module / factory id reference. 

For example: `surfactant-na-0.5-12-1-2020` could be used to specify a simulation run on a certain date using 0.5 % salt concentration and 1 surfactant.

The default `directory` and `top_name` attributes are generated using it, if none are provided, following the format of the `FileTreeBuilder`

Reply:
I've tidied up the file description and docstrings to make this a bit clearer. Also, with the addition of the `GromacsTopologyData` class, some other functionalities have been simplified.